docs/source/loading_images.md

```diff
@@ -0,0 +1,267 @@
+    """
+    Read an image file with scipy and return a torch.Tensor.
+
+    :param input_filename: Source image file path.
+    :return: torch.Tensor of shape (C, H, W).
+    """
+    numpy_array = imageio.imread(input_filename)
+    torch_tensor = TF.to_tensor(numpy_array)
+    return torch_tensor
+```
+
+## SimpleITK
+
+[SimpleITK](https://simpleitk.org/) requires a two step process to load an image and extract the data as a numpy array, but it is then in the correct format.
+
+```python
+from pathlib import Path
+
+import SimpleITK as sitk
+import torch
+import torchvision.transforms.functional as TF
+
+def read_image_sitk(input_filename: Path) -> torch.Tensor:
+    """
+    Read an image file with SimpleITK and return a torch.Tensor.
+
+    :param input_filename: Source image file path.
+    :return: torch.Tensor of shape (C, H, W).
+    """
+    itk_image = sitk.ReadImage(str(input_filename))
+    numpy_array = sitk.GetArrayFromImage(itk_image)
+    torch_tensor = TF.to_tensor(numpy_array)
+    return torch_tensor
+```
+
+## scikit-image
+
+[scikit-image](https://scikit-image.org/) is also very simple to use, since it loads the image as a numpy array in the correct format.
+
+```python
+from pathlib import Path
+
+from skimage import io
+import torch
+import torchvision.transforms.functional as TF
+
+def read_image_skimage(input_filename: Path) -> torch.Tensor:
+    """
+    Read an image file with scikit-image and return a torch.Tensor.
+
+    :param input_filename: Source image file path.
+    :return: torch.Tensor of shape (C, H, W).
+    """
+    numpy_array = io.imread(input_filename)
+    torch_tensor = TF.to_tensor(numpy_array)
+    return torch_tensor
+```
+
+## numpy
+
+For comparison, the png image data was saved in the numpy native data format and then reloaded.
+
+```python
+from pathlib import Path
+
+import numpy as np
+import torch
+import torchvision.transforms.functional as TF
+
+def read_image_numpy(input_filename: Path) -> torch.Tensor:
+    """
+    Read an Numpy file with Torch and return a torch.Tensor.
+
+    :param input_filename: Source image file path.
+    :return: torch.Tensor of shape (C, H, W).
+    """
+    numpy_array = np.load(input_filename)
+    torch_tensor = torch.from_numpy(numpy_array)
+    return torch_tensor
+```
+
+## torch
+
+Again, for comparison, the png image data was saved in the torch tensor native data format and then reloaded.
+
+```python
+from pathlib import Path
+
+import torch
+
+def read_image_torch2(input_filename: Path) -> torch.Tensor:
+    """
+    Read a Torch file with Torch and return a torch.Tensor.
+
+    :param input_filename: Source image file path.
+    :return: torch.Tensor of shape (C, H, W).
+    """
+    torch_tensor = torch.load(input_filename)
+    return torch_tensor
+```
+
+## Results
+
+All the above methods were ran against 122 RGB images, and 122 greyscale images, repeated 10 times. So in total there were 2440 calls to each of the functions. The total times were:
```

Comment:
Are the performance numbers on RGB and greyscale the same? Any change we can have tables for the two image types separately?

Reply:
Ok, I'll profile separately